### ISSUE TITLE
nixos: GNOME Settings Power Mode doesn't do anything?

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#263008]
### Describe the bug
It seems that changing the Power Mode under GNOME Settings -> Power doesn't have any effect on the system.

### Steps To Reproduce
1. Open GNOME Settings and go to Power.
2. Cycle through the Power Modes: Performance/Balanced/Power Saver
3. See no sysfs activity / noticeable battery life changes.

### Expected behavior
I'd expect some process to open/write some scaling_governor sysfs file while manipulating the GUI, which should be visible with either:
* `inotifywait -m -r /sys/ | grep governor`
* `watch -d grep . /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor`

But nothing happens.

### Screenshots
![Screenshot from 2023-10-23 21-10-24](https://github.com/NixOS/nixpkgs/assets/133602/c546ad52-e625-49fd-9c42-77a3625219c4)

### Additional context
The cpupower CLI can change cpu scaling governor: `sudo cpupower frequency-set -g performance` (is observable in sysfs).

### Metadata
NixOS 23.05 (898cb2064b6e98b8c5499f37e81adbdf2925f7c5).
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

**Comment by SuperSamus** [comment]
What's the output of `powerprofilesctl`?

**Comment by bjornfor** [comment]
It was stuck like this:

```
$ powerprofilesctl 
* performance:
    Driver:     intel_pstate
    Degraded:   no

  balanced:
    Driver:     intel_pstate

  power-saver:
    Driver:     intel_pstate
```

(Changing modes in the GUI didn't change this output.)

However, after a reboot it started working!

Some things I've been doing lately that needs revisiting:
* I removed `powerManagement.cpuFreqGovernor = lib.mkDefault "powersave";`, following fb602e5608735f7cb427226f0d2ea539b5129cc2. So after the reboot I did not have that NixOS option set.
* Before the reboot I used `cpupower` to manually change CPU scaling governors. This is the same tool as NixOS uses as backend for `powerManagement.cpuFreqGovernor`. I wonder if using `cpupower` or the NixOS option breaks the GNOME Settings Power Mode GUI/service?

**Comment by SuperSamus** [comment]
> intel_pstate

If you are using [PState drivers](https://www.kernel.org/doc/html/v6.5/admin-guide/pm/intel_pstate.html), then traditional governors will behave in a completely different way.
TL;DR, there is no point in changing the regular governor.

The file you will be interested with in `/sys` (and that `power-profiles-daemon` will change) is `energy_performance_preference`.
It can only be changed if the regular governor is `powersave` ([`power-profiles-daemon` will do that automatically](https://gitlab.freedesktop.org/hadess/power-profiles-daemon/-/blob/main/README.md?ref_type=heads#operations-on-intel-based-machines)).

> However, after a reboot it started working!

You probably set the regular governor to `performance`.

**Comment by SuperSamus** [comment]
If the behavior I described is what happens to you (that is, if everything is working as intended), please close the issue.

**Comment by bjornfor** [comment]
Observing with `watch -n1 -d powerprofilesctl`, the GUI only gets "stuck" after manually running `sudo cpupower frequency-set -g performance`. And it can be "un-stuck" with `sudo cpupower frequency-set -g powersave`.

Since NixOS hasn't ever emitted `powerManagement.cpuFreqGovernor = lib.mkDefault "performance"` in hardware-configuration.nix, this can be closed.
